projectmanager/projectmanager.go

```diff
@@ -0,0 +1,139 @@
+package projectmanager
+
+import (
+	"encoding/json"
+	"fmt"
+	c "github.com/codilime/floodgate/config"
+	gc "github.com/codilime/floodgate/gateclient"
+	spr "github.com/codilime/floodgate/spinnakerresource"
+	"github.com/codilime/floodgate/util"
+	log "github.com/sirupsen/logrus"
+	"os"
+	"path/filepath"
+)
+
+// ProjectManager stores resources and data for specified project and has methods for access, saving.
+type ProjectManager struct {
+	resources         spr.SpinnakerResources
+	client            *gc.GateapiClient
+	projectName       string
+	pipelineConfigIds []string
+}
+
+// Init initializes ProjectManager
+func (pm *ProjectManager) Init(config *c.Config, projectName string) error {
+	pm.projectName = projectName
+	pm.client = gc.NewGateapiClient(config)
+	user, _, err := pm.client.AuthControllerApi.LoggedOutUsingGET(pm.client.Context)
+	if err != nil {
+		return err
+	}
+	if user == "" {
+		return fmt.Errorf("authenticating with Spinnaker failed. check if credentials are valid")
+	}
+
+	if err := pm.getApplications(); err != nil {
+		return err
+	}
+
+	if err := pm.getPipelines(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (pm *ProjectManager) getApplications() error {
+	log.Print("Loading project applications...")
+
+	payload, _, err := pm.client.ProjectControllerApi.GetUsingGET1(pm.client.Context, pm.projectName)
+	if err != nil {
+		return err
+	}
+
+	var projectConfig map[string]interface{}
+	if _, exists := payload["config"]; exists {
+		config, ok := payload["config"].(map[string]interface{})
+		if ok {
+			projectConfig = config
+		}
+	}
+
+	for _, appName := range projectConfig["applications"].([]interface{}) {
```

Comment:
As discussed offline: watch out for assertion exceptions. Ditto to every such assertion in this file, add "if ok { [...] }".